### DISCUSSION TITLE
Naming things: The Proofs formerly known as Fraud

### DISCUSSION THREAD

**Discussion by maurelian** [ethereum-optimism/optimistic-specs#53] [Q&A] [answered]
**TLDR: 'Fraud Proof' is a bad name. Please help us identify a good one.**

## Motivation

'Fraud Proof's (hereafter referred to as _The Mechanism_ are a critical (if not definitive) component of Optimistic Rollups.

One cannot achieve even a basic high level understanding of the ORU design pattern without encountering and discussing them. 

However we've found over time that the negative connotations of the word "Fraud" tend to impede, rather than facilitate a clear understanding of how an optimistic system should work.
Moreover, Fraud implies intent when it is possible for an honest party to submit invalid state roots due to software bugs.

For these reason we're looking for input on a new name for _The Mechanism_, which would be more literally descriptive, and more accessible to the general public. 

Here are some of the ideas we've batted around already: 
* Correctness Challenge
* Transaction Result Challenge
* ~~Transaction Challenge~~  <sup>[1](#footnotes)</sup>
* State Proposal Challenge
* State Transition Challenge
* Execution Challenge
* Rollup State Challenge
* Verification challenge

Clearly we really like the term "Challenge", but that's also up for debate if you have alternative suggestions.

## **Technical Footnotes:**

<a name="footnotes">1. </a> A common misunderstanding even among very knowledgable people is that the challenge applies to a Transaction, and that the transaction can be removed or reverted if it is 'fraudulent'. However that is not the case; all transactions are valid, and can be included in the chain, it is only the proposed result of executing the transaction that can be challenged. (This is why we've already rejected 'Transaction Challenge'). 

2. It's worth noting that the eventual design of _The Mechanism_ will actually challenge state transitions which result from executing multiple transactions.

**Comment by 0xalizk** [comment]
FIST: Foiling Invalid State Transitions

**Comment by smartcontracts** [comment]
I don't think the word "State" should be in any names that we use. It's a confusing concept, especially for people who don't have a background in computer science.

**Comment by smartcontracts** [comment]
My vote is for "Transaction Result Challenge".

It's clear and mostly accurate. You're challenging the published result of one or more transactions.

**Reply by smartcontracts** [reply]
"Transaction Result Challenge" also allows you to easily lead into the point that "you're not challenging *transactions*, you're challenging transaction *results*".

**Reply by maurelian** [reply]
This is probably the most direct AND accessible (more people understand 'transaction' than 'state'). 
How would you translate this sentence from the current nomenclature?

> The fraudulent state transition occurred at step 20 of the execution...

**Reply by A2be** [reply]
The state transition that occurred at step 20 of the execution was challenged, and the transaction result was reverted  since the challenge demonstrated xyz about the initial result on the Optimism Layer 2 chain.

**Comment by Nicca42** [comment]
* Execution Validation
* Batch Validation
* Correctness Proof 
* Validity Challenge 
* Execution Result Challenge (running on from @smartcontracts answer, removes possible confusion of it being about challenging transactions)

**Reply by maurelian** [reply]
Thanks! I think among these 'Correctness Proof' (or 'Challenge') would be my favorite. Correct/incorrect are quite accessible, and it also provide an obvious answer to my question [here](https://github.com/ethereum-optimism/optimistic-specs/discussions/53#discussioncomment-1761871), ie. we'd say

> The incorrect state transition occurred at step 20 of the execution...

**Comment by norswap** [comment, marked as answer]
What about "*fault proof*"?

Advantages:
1. rolls off the tongue, short
2. sounds similar to "fraud proof"
3. not misleading: doesn't seem to entail something about transactions being invalid, or wrong state roots being necessarily intentional

These may small things, but I think they matter.

Regarding (1), I think "transaction result challenge" is clearest... but also it's a mouthful, people aren't going to bother typing it out. A vote for "transaction result challenge" is a vote for "TRC" because that's what people will end up using.

Regarding (2), *fraud proof* is well established and people are going to continue using it. The real reason we're changing is (a) because fraud implies intent and there isn't always intent behind a wrong state being computed (could be a bug), (b) legal implications of the word "fraud". I think *fault proof* solves both issues while sounding so similar to fraud proof that people will immediately make the connection.

**Reply by maurelian** [reply]
The more I think about this the more I like it.

**Reply by smartcontracts** [reply]
Yeah, I think this is the best one yet.

**Reply by maurelian** [reply]
Ah, well if you're coming around, and @stonecoldpat [agrees](https://twitter.com/stonecoldpat0/status/1468216689183674370), I think we've got a winner.

**Reply by jennypollack** [reply]
froof

**Reply by transmissions11** [reply]
slaps

**Comment by Dominator008** [comment]
Invalidity Proof? In contrast to validity proofs.

**Comment by kethcode** [comment]
Well, the Proof is a list of valid transactions. Transactions not in the list have not been confirmed valid. The list itself also needs to be internally consistent. So it's more like an audit than anything else. 

So, Validity Audit, perhaps? Block Audit?  Fault Audit?

In general usage, people will be refering to 'It' as valid or invalid, so shouldn't use that word.  It's a valid audit, or an invalid audit.  If Audit is too accounting-specific, we could trying some similar words, but this is where I'm landing right now.

-----

Just read up.  Fault Proof is pretty good.  Similar feeling, and retails some familiar terminology.    I could see Fault Proof or Fault Audit working well.
